src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -520,6 +540,14 @@ protected override AuthenticationTicket AuthenticateCore()
             }
         }
 
+        protected virtual IdentityModel.Clients.ActiveDirectory.AuthenticationResult RedeemAuthorizationCode(string authorizationCode)
```

Comment:
God no, not this mess. You can't even imagine how many people lost their hair on the #owin channel because of token caching :smile:

And why would you use caching here? New code, new token. Absolutely no need for caching.

@vibronet hey, it's been a long time... :smile: